admin/election_controller.go

```diff
@@ -0,0 +1,153 @@
+	store model.ElectionStore
+}
+
+// NewElectionController creates a new election controller
+func NewElectionController(store model.ElectionStore) *ElectionController {
+	return &ElectionController{store: store}
+}
+
+// Create creates a new election with it's users
+func (c *ElectionController) Create(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	newElection := model.Election{}
+	if err := json.Unmarshal(msg.Data, &newElection); err != nil {
+		return err
+	}
+
+	election, err := c.store.CreateElection(&newElection)
+	if err != nil {
+		return err
+	}
+
+	electionBearerToken, err := GenerateAdminToken(election.ID)
+	if err != nil {
+		return err
+	}
+
+	response := ElectionWithTokenResponse{
+		AdminToken: electionBearerToken,
+		Election:   *election,
+	}
+
+	return ctx.Send(new(ApiResponse).Set(response).MustMarshall(), apirest.HTTPstatusOK)
+}
+
+// Election returns the election data
+func (c *ElectionController) Election(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	var electionID types.HexBytes
+	if err := electionID.FromString(strings.TrimPrefix(ctx.URLParam("electionId"), "0x")); err != nil {
+		return err
+	}
+	election, err := c.store.Election(electionID)
+	if err != nil {
+		return err
+	}
+
+	return ctx.Send(new(ApiResponse).Set(election).MustMarshall(), apirest.HTTPstatusOK)
+}
+
+// Delete deletes an election
+func (c *ElectionController) Delete(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	var electionID types.HexBytes
+	if err := electionID.FromString(strings.TrimPrefix(ctx.URLParam("electionId"), "0x")); err != nil {
+		return err
+	}
+
+	valid, err := ValidateAdminToken(electionID, msg.AuthToken)
+	if !valid || err != nil {
+		return ctx.Send(
+			new(ApiResponse).SetError(CodeErrInvalidAuth, ReasonErrInvalidAuth).MustMarshall(),
+			apirest.HTTPstatusBadRequest,
+		)
+	}
+
+	if err := c.store.DeleteElection(electionID); err != nil {
+		return err
+	}
+
+	return ctx.Send(new(ApiResponse).Set(nil).MustMarshall(), apirest.HTTPstatusOK)
+}
+
+// List returns the list of elections
+func (c *ElectionController) List(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	elections, err := c.store.ListElection()
+	if err != nil {
+		return err
+	}
+
+	return ctx.Send(new(ApiResponse).Set(elections).MustMarshall(), apirest.HTTPstatusOK)
+}
+
+type AdminTokenRequest struct {
+	Signature string `json:"signature"`
+	Data      string `json:"data"`
+}
+
+// AdminToken returns the admin token for an election
+func (c *ElectionController) AdminToken(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	var electionID types.HexBytes
+	if err := electionID.FromString(strings.TrimPrefix(ctx.URLParam("electionId"), "0x")); err != nil {
+		return err
+	}
+
+	adminTokenRequest := AdminTokenRequest{}
+	if err := json.Unmarshal(msg.Data, &adminTokenRequest); err != nil {
+		return err
+	}
+
+	signature, _ := hexStringToBytes(strings.TrimPrefix(adminTokenRequest.Signature, "0x"))
```

Comment:
A non checked error would make the CI cry... so please don't do it :+1:

Reply:
You are doing `strings.TrimPrefix(data, "0x")` many times in the code.
Consider abstracting this into hexStringToBytes()